src/main/java/com/wiley/autotest/selenium/elements/upgrade/BaseTeasyElement.java

```diff
@@ -0,0 +1,495 @@
+        }
+    }
+
+    private Object executeScript(String script, Object... args) {
+        return ((JavascriptExecutor) getWebDriver()).executeScript(script, args);
+    }
+
+    private List<WebElement> finds(By by) {
+        try {
+            //for catch stale element
+            isEnabled();
+            return (getFrameTransparentDriver()).findElements(this, by);
+        } catch (StaleElementReferenceException e) {
+            againLocate();
+            return finds(by);
+        } catch (UndeclaredThrowableException e) {
+            //This checks for findElementByNoThrow, otherwise we call againLocate
+            if (((InvocationTargetException) e.getUndeclaredThrowable()).getTargetException() instanceof NoSuchElementException) {
+                try {
+                    return (getFrameTransparentDriver()).findElements(this, by);
+                } catch (UndeclaredThrowableException noSuchElementException) {
+                    throw new NoSuchElementException("Unable to find elements " + by.toString() + ", Exception - " + ((InvocationTargetException) noSuchElementException
+                            .getUndeclaredThrowable()).getTargetException().getMessage());
+                }
+            }
+            againLocate();
+            return finds(by);
+        }
+    }
+
+    private WebElement find(By by) {
+        try {
+            //for catch stale element
+            isEnabled();
+            return (getFrameTransparentDriver()).findElement(this, by);
+        } catch (StaleElementReferenceException e) {
+            againLocate();
+            return find(by);
+        } catch (UndeclaredThrowableException e) {
+            //This checks for findElementByNoThrow, otherwise we call againLocate
+            if (((InvocationTargetException) e.getUndeclaredThrowable()).getTargetException() instanceof NoSuchElementException) {
+                try {
+                    return getFrameTransparentDriver().findElement(this, by);
+                } catch (UndeclaredThrowableException noSuchElementException) {
+                    throw new NoSuchElementException("Unable to find element " + by.toString() + ", Exception - " + ((InvocationTargetException) noSuchElementException
+                            .getUndeclaredThrowable()).getTargetException().getMessage());
+                }
+            }
+            againLocate();
+            return find(by);
+        } catch (NoSuchElementException e) {
+            return (getFrameTransparentDriver()).findElement(this, by);
+        } catch (WebDriverException e) {
+            againLocate();
+            return find(by);
+        }
+    }
+
+    private WebElement getParentElement(WebElement element) {
+        return (WebElement) executeScript("return arguments[0].parentNode", element);
+    }
+
+    private FramesTransparentWebDriver getFrameTransparentDriver() {
+        return (FramesTransparentWebDriver) getWebDriver();
+    }
+
+}
```

Comment:
Class looks too big

Reply:
@dpavlov I absolutely agree. However, I don't see how I can reduce the number of public methods. But I will try to encapsulate some bulky logic to the objects.